src/main/java/net/fabricmc/loader/util/mappings/MixinIntermediaryDevRemapper.java

```diff
@@ -16,46 +16,47 @@
 
 package net.fabricmc.loader.util.mappings;
 
-import net.fabricmc.mappings.*;
-import net.fabricmc.mappings.helpers.mixin.MixinMappingsRemapper;
+import net.fabricmc.mapping.tree.ClassDef;
+import net.fabricmc.mapping.tree.Descriptored;
+import net.fabricmc.mapping.tree.TinyTree;
+import net.fabricmc.mapping.util.MixinRemapper;
 import org.spongepowered.asm.mixin.transformer.ClassInfo;
 
 import java.util.*;
 
-public class MixinIntermediaryDevRemapper extends MixinMappingsRemapper {
+public class MixinIntermediaryDevRemapper extends MixinRemapper {
 	private final Set<String> allPossibleClassNames;
 	private final Map<String, Set<String>> nameDescFieldLookup, nameDescMethodLookup;
 
-	private static String getNameDescKey(EntryTriple triple) {
-		return triple.getName() + ";;" + triple.getDesc();
+	private static String getNameDescKey(String name, String descriptor) {
+		return name+ ";;" + descriptor;
 	}
 
-	public MixinIntermediaryDevRemapper(Mappings mappings, String from, String to) {
+	public MixinIntermediaryDevRemapper(TinyTree mappings, String from, String to) {
 		super(mappings, from, to);
 
-		// I sincerely hate that I have to do this.
+		// Asie sincerely hated that he had to do this.
```

Comment:
![](https://i.imgur.com/zJEiJ4k.gif)

We can probably just get rid of the comment.

Reply:
preposterous

Reply:
I sincerely hate that fudge asks me not to make the remapper final because the loader extends that remapper from the mapping parser

Reply:
hey blame @asiekierka for making it like that in the first place, i'm not gonna rewrite loader.

Reply:
This whole class is a hack. If you can remove this hack, all power to you.

Reply:
you can just feed the whole set of mapping to this remapper I think